File: t/CS/DicomDigitalSignatureDemo/UI/CreateSignature.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;
using Leadtools.Demos;

namespace DicomDigitalSignatureDemo.UI
{
   public partial class CreateSignature : Form
   {
      public string PrivateKey
      {
         get
         {
            return _txtBx_PrivateKey.Text;
         }
      }
      public string DigitalCertificate
      {
         get
         {
            return _txtBx_DigitalCertificate.Text;
         }
      }
      public string Password
      {
         get
         {
            return _txtBx_Password.Text;
         }
      }
      public string SignedDataSet
      {
         get
         {
            return _txtBox_SignedDataSet.Text;
         }
      }
      public CreateSignature()
      {
         InitializeComponent();

         string certificate = GetFullCertificatePath();

         if (certificate != null)
         {
            _txtBx_DigitalCertificate.Text = certificate;
            _txtBx_PrivateKey.Text = certificate;
            _txtBx_Password.Text = "test";
         }

         checkIfOKEnabled();
      }

      private void _btn_PrivateKey_Click(object sender, EventArgs e)
      {
         OpenFileDialog fileDialog=new OpenFileDialog();
         fileDialog.Filter="All Files (*.*)|*.*";

         if (fileDialog.ShowDialog() == DialogResult.OK)
         {
            this._txtBx_PrivateKey.Text = fileDialog.FileName;
         }
      }

      private void _btn_DigitalCertificate_Click(object sender, EventArgs e)
      {
         OpenFileDialog fileDialog = new OpenFileDialog();
         fileDialog.Filter = "Digital Certificates (*.pem; *.cer; *.crt; *.p7b; *.spc; *.pfx; *.p12)|" +
                              "*.pem; *.cer; *.crt; *.p7b; *.spc; *.pfx; *.p12|" +
                              "All Files (*.*)|*.*";

         if (fileDialog.ShowDialog() == DialogResult.OK)
         {
            this._txtBx_DigitalCertificate.Text = fileDialog.FileName;
         }
      }

      private void _btn_SignedDataSet_Click(object sender, EventArgs e)
      {
         SaveFileDialog fileDialog = new SaveFileDialog();
         fileDialog.Filter = "DCM Files (*.dcm)|*.dcm|All Files (*.*)|*.*";

         if (fileDialog.ShowDialog() == DialogResult.OK)
         {
            this._txtBox_SignedDataSet.Text = fileDialog.FileName;
         }
      }

      private string GetFullCertificatePath()
      {
         string path = Path.GetDirectoryName(Application.ExecutablePath);
         path = Path.Combine(path, "client.pem");
         if (!File.Exists(path))
         {
            path = Path.Combine(DemosGlobal.ImagesFolder, "client.pem");
         }

         if (File.Exists(path))
            return path;
         return null;
      }

      private void _txtBx_PrivateKey_TextChanged(object sender, EventArgs e)
      {
         checkIfOKEnabled();
      }

      private void _txtBx_DigitalCertificate_TextChanged(object sender, EventArgs e)
      {
         checkIfOKEnabled();
      }

      private void _txtBox_SignedDataSet_TextChanged(object sender, EventArgs e)
      {
         checkIfOKEnabled();
      }

      void checkIfOKEnabled() 
      {
         if (_txtBx_PrivateKey.Text == "" || _txtBx_DigitalCertificate.Text == "" || _txtBox_SignedDataSet.Text == "")
            _btn_OK.Enabled = false;
         else
            _btn_OK.Enabled = true;
      }
   }
}

File: t/VB/DicomDigitalSignatureDemo/UI/CreateSignature.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.IO
Imports System

Namespace DicomDigitalSignatureDemo.UI
   Partial Public Class CreateSignature
      Inherits Form
      Public ReadOnly Property PrivateKey() As String
         Get
            Return _txtBx_PrivateKey.Text
         End Get
      End Property
      Public ReadOnly Property DigitalCertificate() As String
         Get
            Return _txtBx_DigitalCertificate.Text
         End Get
      End Property
      Public ReadOnly Property Password() As String
         Get
            Return _txtBx_Password.Text
         End Get
      End Property
      Public ReadOnly Property SignedDataSet() As String
         Get
            Return _txtBox_SignedDataSet.Text
         End Get
      End Property
      Public Sub New()
         InitializeComponent()

         Dim certificate As String = GetFullCertificatePath()

         If certificate IsNot Nothing Then
            _txtBx_DigitalCertificate.Text = certificate
            _txtBx_PrivateKey.Text = certificate
            _txtBx_Password.Text = "test"
         End If

         checkIfOKEnabled()
      End Sub

      Private Sub _btn_PrivateKey_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btn_PrivateKey.Click
         Dim fileDialog As New OpenFileDialog()
         fileDialog.Filter = "All Files (*.*)|*.*"

         If fileDialog.ShowDialog() = DialogResult.OK Then
            Me._txtBx_PrivateKey.Text = fileDialog.FileName
         End If
      End Sub

      Private Sub _btn_DigitalCertificate_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btn_DigitalCertificate.Click
         Dim fileDialog As New OpenFileDialog()
         fileDialog.Filter = "Digital Certificates (*.pem; *.cer; *.crt; *.p7b; *.spc; *.pfx; *.p12)|" & "*.pem; *.cer; *.crt; *.p7b; *.spc; *.pfx; *.p12|" & "All Files (*.*)|*.*"

         If fileDialog.ShowDialog() = DialogResult.OK Then
            Me._txtBx_DigitalCertificate.Text = fileDialog.FileName
         End If
      End Sub

      Private Sub _btn_SignedDataSet_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btn_SignedDataSet.Click
         Dim fileDialog As New SaveFileDialog()
         fileDialog.Filter = "DCM Files (*.dcm)|*.dcm|All Files (*.*)|*.*"

         If fileDialog.ShowDialog() = DialogResult.OK Then
            Me._txtBox_SignedDataSet.Text = fileDialog.FileName
         End If
      End Sub

      Private Function GetFullCertificatePath() As String
         Dim path__1 As String = Path.GetDirectoryName(Application.ExecutablePath)
         path__1 = Path.Combine(path__1, "client.pem")
         If File.Exists(path__1) Then
            Return path__1
         End If
         Return Nothing
      End Function

      Private Sub _txtBx_PrivateKey_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles _txtBx_PrivateKey.TextChanged
         checkIfOKEnabled()
      End Sub

      Private Sub _txtBx_DigitalCertificate_TextChanged(ByVal sender As Object, ByVal e As EventArgs)
         checkIfOKEnabled()
      End Sub

      Private Sub _txtBox_SignedDataSet_TextChanged(ByVal sender As Object, ByVal e As EventArgs)
         checkIfOKEnabled()
      End Sub

      Private Sub checkIfOKEnabled()
         If _txtBx_PrivateKey.Text = "" OrElse _txtBx_DigitalCertificate.Text = "" OrElse _txtBox_SignedDataSet.Text = "" Then
            _btn_OK.Enabled = False
         Else
            _btn_OK.Enabled = True
         End If
      End Sub
   End Class
End Namespace
